message/message1_1prime/transfer_message.go

```diff
@@ -0,0 +1,44 @@
+package message1_1
+
+import (
+	"io"
+
+	"github.com/ipld/go-ipld-prime/codec/dagcbor"
+	"github.com/ipld/go-ipld-prime/datamodel"
+	"github.com/ipld/go-ipld-prime/node/bindnode"
+	"github.com/ipld/go-ipld-prime/schema"
+
+	datatransfer "github.com/filecoin-project/go-data-transfer"
+)
+
+// TransferMessage1_1 is the transfer message for the 1.1 Data Transfer Protocol.
+type TransferMessage1_1 struct {
+	IsRequest bool
+
+	Request  *TransferRequest1_1
+	Response *TransferResponse1_1
+}
+
+// ========= datatransfer.Message interface
+
+// TransferID returns the TransferID of this message
+func (tm *TransferMessage1_1) TransferID() datatransfer.TransferID {
+	if tm.IsRequest {
+		return tm.Request.TransferID()
+	}
+	return tm.Response.TransferID()
+}
+
+func (tm *TransferMessage1_1) toIPLD() schema.TypedNode {
+	return bindnode.Wrap(&tm, Prototype.TransferMessage.Type())
```

Comment:
just noting you have about 6x places return value here is ultimately used and all of them call .Representation() immediately. I wonder if you  just return the representation node and save yourself the trouble.

Reply:
Yes .. the original version of my code does have it, but then there was one place I found I wanted it back to a TypedNode.. Again I think some of the serialisation stuff will change so maybe this can go back. At least in the current state of my next PR this method has entirely changed and bindnode logic has been moved elsewhere .. but that's in flux while I work my head through the rest of the system.